host_core/native/hostcore_wasmcloud_native/src/wasmruntime.rs

```diff
@@ -38,111 +48,109 @@ pub struct ExRuntimeConfig {
     host_id: String,
 }
 
+#[derive(Clone)]
 pub struct ElixirHandler {
-    // Runtime pid
+    /// Runtime process ID
     pid: LocalPid,
-    #[allow(unused)]
-    host_id: String,
+    claims: jwt::Claims<jwt::Actor>,
+    call_context: Vec<u8>,
+}
+
+#[derive(Clone)]
+struct ElixirHandlerArc(Arc<ElixirHandler>);
```

Comment:
I know naming is hard but I'd rather not put a data type like arc as a suffix on a struct.

Reply:
Could you suggest a better option?

Reply:
It was called ElixirHandler before. The fact that the inner element is an arc shouldn't be advertised in the name.

Reply:
But what would the inner struct be called then? `ElixirHandlerInner`? I really don't see how that's better.

The only option to avoid this is to store an unnamed tuple in an Arc, but that'd be very difficult to maintain and use

Reply:
I'm also not sure what caused the need for the arc wrapper around the elixir handler. Since I'm on a phone deferring to Taylor for thoughts on the abstraction and name

Reply:
> not sure what caused the need for the arc wrapper around the elixir handler

Mainly, wasmtime/wit-bindgen design drawbacks, namely the fact that the host trait implementations take `&mut self` parameter, therefore forcing this pattern on us

Reply:
But shouldn't we be insulating consumers of our rust wasmCloud library from design changes and decisions of the underlying libraries? We shouldn't have to refactor the otp nif every time wasmtime makes an internal change... Isolating the host from this kind of churn is actually the primary purpose of the rust wasmCloud library.

Reply:
> > not sure what caused the need for the arc wrapper around the elixir handler
> 
> Mainly, wasmtime/wit-bindgen design drawbacks, namely the fact that the host trait implementations take `&mut self` parameter, therefore forcing this pattern on us

If they force this on our wasmCloud library then the wasmCloud library should not pass this along to its consumers. It is designed specifically to prevent this

Reply:
Sure, eventually we should think about this, depending on the state of things in the future, right now it's way too early to build these abstractions.

Reply:
I disagree. Our job with libwasmcloud is to isolate consumers from wasmtime and its volatility. It's our job to come up with a more stable abstraction than the raw wasmtime interface.